c10/core/SymBool.cpp

```diff
@@ -1,84 +1,79 @@
+#include <c10/core/ConstantSymNodeImpl.h>
 #include <c10/core/SymBool.h>
 #include <c10/core/SymNodeImpl.h>
 #include <array>
 #include <utility>
 
 namespace c10 {
 
+SymNode SymBool::promoteToConstantSymNode() const {
+  TORCH_CHECK(!is_heap_allocated())
+  return SymNode(c10::make_intrusive<ConstantSymNodeImpl<bool>>(data_));
```

Comment:
I don't really understand why you need this. Can you tell me more?

Reply:
Good point, we actually we don't. I can just do .expect_bool() instead